### PR TITLE
[FIVE-266] (SignIn) Incorporación de ServiceResponse en SignInService y su controller

### PR DESCRIPTION
**Pull request by martinmena**
## Cambios realizados
- Se creó un nuevo tipo de error en ErrorCodes para representar el error de autenticación.
- Se modificó la interfaz del servicio de SignIn para que su método devuelva un objeto ServiceResponse.
- Se modificó el método del servicio de SignIn para que las posibles respuestas sean dicho objeto, reemplazando el uso de la Tupla.
- Se modificaron los tests acorde a los cambios en el servicio de SignIn y el uso de objetos ServiceResponse.
- Se modificó el método del controller de SignIn para adaptarlo a las nuevas respuestas del servicio.

### CODE REVIEW

**Review by matiasbeckerle — APPROVED** [review]
💯

**Review by gperelstein — APPROVED** [review]

File: FiveRockingFingers/FRF.Core/Services/SignInService.cs [hunk 1]
```diff
@@ -30,13 +31,13 @@ public SignInService(SignInManager<CognitoUser> signInManager)
...
-                if (token == null) return new Tuple<bool, string>(false, "");
+                if (token == null) return new ServiceResponse<string>(new Error(ErrorCodes.AuthentificationError, ""));
 
                 var result = await _signInManager.PasswordSignInAsync(token, userPassword,
                     userSignIn.RememberMe, lockoutOnFailure: false);
                 return result.Succeeded
-                    ? new Tuple<bool, string>(true, token.UserID)
-                    : new Tuple<bool, string>(false, "");
+                    ? new ServiceResponse<string>(token.UserID)
+                    : new ServiceResponse<string>(new Error(ErrorCodes.AuthentificationError, ""));
```
Comment (gonzalorom) [inline FiveRockingFingers/FRF.Core/Services/SignInService.cs:40]: Another technical debt solved. Good job mate!

**Review by gonzalorom — APPROVED** [review]

File: FiveRockingFingers/FRF.Core/Response/ErrorCodes.cs [hunk 2]
```diff
@@ -11,5 +11,8 @@ public static class ErrorCodes
         // Relation errors
         public const int RelationAlreadyExisted = 10;
         public const int RelationNotValid = 11;
+
+        // User errors
+        public const int AuthenticationError = 20;
```
Comment (gperelstein) [inline FiveRockingFingers/FRF.Core/Response/ErrorCodes.cs:16]: After riding the conversation in #83 I was thinking if some of the comments there could be implemented here? What do you think?

Comment (martinmena) [reply FiveRockingFingers/FRF.Core/Response/ErrorCodes.cs:16]: Do you mean the 2 new type of errors? InvalidCredentials and AuthenticationServerCurrentlyUnavailable?

Comment (gperelstein) [reply FiveRockingFingers/FRF.Core/Response/ErrorCodes.cs:16]: Exactly

Comment (martinmena) [reply FiveRockingFingers/FRF.Core/Response/ErrorCodes.cs:16]: I think I would rather keep them separated, specially if they already have their own PR. One type of error refers to a user trying to authenticate in the system (sign in, the one in this PR), and the other two are related to someone trying to create a new account (sign up, the ones in #83).
Having said that, I was going to make sure they end up under the same category `// User errors`

Comment (martinmena) [reply FiveRockingFingers/FRF.Core/Response/ErrorCodes.cs:16]: Ohhh I get it now, you're saying that we can do with just 2 errors, keep as it is in the other PR, but use them here also, removing the AuthenticationError right?

Comment (gperelstein) [reply FiveRockingFingers/FRF.Core/Response/ErrorCodes.cs:16]: Yes, exactly! I apologize if I don't explain myself very well

**Pull request merged**